peer.go

```diff
@@ -37,6 +38,22 @@ type baseSession struct {
 
 	client     Peer
 	serializer serializers.Serializer
+
+	session     *Session
+	publishLogs atomic.Bool
+	topic       string
+}
+
+func (b *baseSession) EnableLogPublishing(session *Session, topic string) {
+	if b.session == nil {
```

Comment:
there is a race condition here. both session and topic need to be set using a lock